backend/ownerSizeTotalWorker.test.js

```diff
@@ -0,0 +1,288 @@
+    this.totalSize += difference
+  }
+
+  get kvSize () {
+    return this._kvSize
+  }
+
+  set kvSize (value) {
+    const difference = value - this._kvSize
+    this._kvSize = value
+    this.ownSize += difference
+  }
+
+  get totalSize () {
+    return this._totalSize
+  }
+
+  set totalSize (value) {
+    const difference = value - this._totalSize
+    this._totalSize = value
+    if (this.ultimateOwner) {
+      this.ultimateOwner.totalSize += difference
+    }
+  }
+
+  get ultimateOwner () {
+    if (!this.owner) return null
+    return this.owner.ultimateOwner || this.owner
+  }
+
+  set ultimateOwner (value) {
+    // empty
+  }
+
+  static async saveIndirectResourcesIndex (resourceID: string) {
+    const ownerID = await sbp('chelonia.db/get', `_private_owner_${resourceID}`)
+    let indirectOwnerID = ownerID
+    while ((indirectOwnerID = await sbp('chelonia.db/get', `_private_owner_${indirectOwnerID}`))) {
+      await appendToIndexFactory(`_private_indirectResources_${indirectOwnerID}`)(resourceID)
+    }
+  }
+
+  static registerBillableEntity = appendToIndexFactory('_private_billable_entities')
+
+  static async saveOwner (ownerID: string, resourceID: string) {
+    // Store the owner for the current resource
+    // Use a queue to check that the owner exists, preventing the creation of
+    // orphaned resources (e.g., because the owner was just deleted)
+    await sbp('chelonia.db/set', `_private_owner_${resourceID}`, ownerID)
+    const resourcesKey = `_private_resources_${ownerID}`
+    // Store the resource in the resource index key
+    // This is done in a queue to handle several simultaneous requests
+    // reading and writing to the same key
+    await appendToIndexFactory(resourcesKey)(resourceID)
+    setTimeout(() => this.saveIndirectResourcesIndex(resourceID), (0, Math.random)() * 500 | 0)
+  }
+
+  async init () {
+    await updateSize(this.id, `_private_size_${this.id}`, this.ownSize)
+    if (this.owner) {
+      await this.constructor.saveOwner(this.owner.id, this.id)
+    } else {
+      await this.constructor.registerBillableEntity(this.id)
+    }
+  }
+
+  async newMessage () {
+    const size = (0, Math.random)() * 9999 | 0
+    this.ownSize += size
+    await updateSize(this.id, `_private_size_${this.id}`, size)
+  }
+
+  async newKvValue () {
+    const val = (0, Math.random)() * 9999 | 0
+    const sign = (0, Math.random)()
+    const size = (sign < 0.5) ? Math.max(-this.kvSize, -val) : val
+    this.kvSize += size
+    await updateSize(this.id, `_private_size_${this.id}`, size)
+  }
+}
+
+async function * randomOp (iterations) {
+  const mainContracts: Contract[] = []
+  const allContracts: Contract[] = []
+
+  {
+    const contract = new Contract()
+    console.log('Created initial contract', contract.id)
+    await contract.init()
+    mainContracts.push(contract)
+    allContracts.push(contract)
+    yield
+  }
+
+  while (iterations-- > 0) {
+    const op = (0, Math.random)() * 9 | 0
```

Comment:
All of these `(0, Math.random)` calls, could you convert them to `Math.random()` please? Unless they are required, in which case at the top of this first one could you add a comment explaining why it is required?

Reply:
Ah, am familiar with those kinds of warnings. I don't think any of our linters give that warning though, and the static analysis warnings are dumb, they are the in the wrong, so IMO we shouldn't modify our code to be weirder to appease them as it violates KISS to violate a dumb linter.

Reply:
Not sure, I can make the change if you'd like me to, but IMHO those warnings _are_ useful. I'd certainly look closely any time someone uses `Math.random` (or similar things that are unsafe to use).

Reply:
Yes, I'm not saying the warnings can't be useful, just that most of the time they're written on code that isn't doing anything wrong. Would rather the code be readable without weirdness 🙏